cdap-proto/src/main/java/io/cdap/cdap/proto/DeployEventType.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.proto;
+
+/**
+ *  The deploy event type that is used for capturing metrics.
+ */
+public enum DeployEventType {
+  NEW,
+  EDIT,
+  UPGRADE
```

Comment:
Is the UPGRADE type set/used right now?

Reply:
no it is not used, but I added it for completeness

Reply:
this is a bit of a confusing enum, since upgrade is not used even though it is a metric and it is in this enum.

Instead of this enum, I think it would be better if addApplication returned the edit number. You can change updateApplicationEdit() in AppMetadataStore to return the edit number and keep returning it all the way up. Then emit the metric based on if edit number is 1 or more

Reply:
Should we keep the enum as per https://github.com/cdapio/cdap/pull/14956/files#r1119425214 ?

Reply:
that should be fine, it makes sense if it's used in a place like that